applications/GeoMechanicsApplication/tests/cpp_tests/test_t_normal_flux_condition.cpp

```diff
@@ -43,31 +44,54 @@ void TestTnormalFluxCondition(ModelPart& rModelPart)
     for (unsigned int i = 0; i < equation_id_vector.size(); i++) {
         KRATOS_EXPECT_EQ(equation_id_vector[i], i);
     }
+
+    Matrix left_hand_side_matrix = ZeroMatrix(1, 1);
+    Vector right_hand_side_vector = ZeroVector(1);
+
+    p_condition->CalculateLocalSystem(
+        left_hand_side_matrix, right_hand_side_vector, r_current_process_info);
+
+    for (unsigned int i = 0; i < right_hand_side_vector.size(); i++) {
+        KRATOS_EXPECT_NEAR(right_hand_side_vector[i], RHS[i], 1.0e-4);
+    }
 }
 
-void GenerateTnormalFluxCondition2D2N(ModelPart& rModelPart)
+void AssignNormalHeatFlux(ModelPart& rModelPart, double value)
 {
-    rModelPart.GetProcessInfo()[DOMAIN_SIZE] = 2;
+    rModelPart.GetProcessInfo()[DELTA_TIME] = 1.0;
 
-    Properties::Pointer p_cond_prop = rModelPart.CreateNewProperties(0);
+    for (unsigned int i = 0; i < rModelPart.NumberOfNodes(); i++) {
+        auto p_node = rModelPart.pGetNode(i + 1);
+        p_node->FastGetSolutionStepValue(NORMAL_HEAT_FLUX) = value;
+        p_node->FastGetSolutionStepValue(NORMAL_HEAT_FLUX, 1) = 0;
+    }
+}
+
+void GenerateTnormalFluxCondition2D2N(ModelPart& rModelPart)
+{
     // Geometry creation
     rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
     rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+
+    AssignNormalHeatFlux(rModelPart, 10.0);
+
     std::vector<ModelPart::IndexType> cond_nodes{1, 2};
     rModelPart.CreateNewCondition("TNormalFluxCondition2D2N", 1, cond_nodes,
-                                p_cond_prop);
+                                  rModelPart.CreateNewProperties(0));
 }
 
-KRATOS_TEST_CASE_IN_SUITE(EquationIdVectorTnormalFluxCondition2D2N, KratosGeoMechanicsFastSuite)
+KRATOS_TEST_CASE_IN_SUITE(TNormalFluxCondition2D2N, KratosGeoMechanicsFastSuite)
 {
     Model this_model;
     ModelPart& model_part = this_model.CreateModelPart("Main", 3);
     // Variables addition
     model_part.AddNodalSolutionStepVariable(TEMPERATURE);
+    model_part.AddNodalSolutionStepVariable(NORMAL_HEAT_FLUX);
 
     GenerateTnormalFluxCondition2D2N(model_part);
 
-    TestTnormalFluxCondition(model_part);
+    std::vector<double> rhs{5.0, 5.0};
```

Comment:
This can be const vector and maybe the naming 'expected_right_hand_side' is a bit more clear? Same holds for the other tests

Reply:
Thank you. Just added const

Reply:
Thanks!